Add active admin

EvgeniiMaiorov opened the pull request:
Where?
https://trello.com/c/KazR4z1W/26-add-active-admin

What?
We need to add activeadmin to create admins

How?
Add activeadmin gem
Add backend logic for admin
Update routes
Update frontend logic
Add uniqe validations for interests and interest_categories names
Add uniqe index name for interests and respective migrations

Commit:
Add active admin

Where?
https://trello.com/c/KazR4z1W/26-add-active-admin

What?
We need to add activeadmin to create admins

How?
Add activeadmin gem
Add backend logic for admin
Update routes
Update frontend logic
Add uniqe validations for interests and interest_categories names
Add uniqe index name for interests and respective migrations
Fix active admin redirection

anevsevra commented on config/application.rb:
```diff
@@ -12,7 +12,8 @@ module SocialNetwork
     # Initialize configuration defaults for originally generated Rails version.
     config.load_defaults 6.0
-
+    config.api_only = false
+    config.session_store = :memory_store
```
Is it required? `session_store` is cookies by default. Should work for html formats.

anevsevra commented on app/assets/stylesheets/application.css:
```diff
@@ -10,7 +10,6 @@
  * files in this directory. Styles in this file should be added after the last require_* statement.
  * It is generally better to create a new file per style scope.
  *
- *= require_tree .
```
This prevents other css from current directory to be loaded.

anevsevra commented on app/controllers/pages_controller.rb:
```diff
@@ -2,4 +2,10 @@
 class PagesController < ApplicationController
   def index; end
+
+  def login_as
```
Move to separate controller with `authenticate_admin_user!` before action.

Commit:
Fix code after review

What?
Since * = require_tree . prevents other CSS from being loaded from the current directory we need to add require_tree . back
Move logi as to separate controller with authenticate_admin_user! before action.

How?
Return * = require_tree .
Add * = stub "active_admin" to avoid some of the styling changes that active admin adds
Add authentications controller
Update routes
